svc-events/events/subscriptionCache.go

```diff
@@ -0,0 +1,210 @@
+package events
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	l "github.com/ODIM-Project/ODIM/lib-utilities/logs"
+	"github.com/ODIM-Project/ODIM/svc-events/evmodel"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	subscriptionsCache                    = make(map[string]evmodel.Subscription)
+	systemToSubscriptionsMap              = make(map[string]map[string]bool)
+	aggregateIdToSubscriptionsMap         = make(map[string]map[string]bool)
+	collectionToSubscriptionsMap          = make(map[string]map[string]bool)
+	emptyOriginResourceToSubscriptionsMap = make(map[string]map[string]bool)
+	systemIdToAggregateIdsMap             = make(map[string]map[string]bool)
+	eventSourceToManagerIDMap             = make(map[string]string)
+	managerIDToSystemIDsMap               = make(map[string][]string)
+	managerIDToChassisIDsMap              = make(map[string][]string)
+)
+
+func LoadSubscriptionData() {
+	l.Log.Debug("Event data load initialized ")
+	t := time.Now()
+	defer l.Log.Debug("Time take to read Complete LoadSubscriptionData ", time.Since(t))
+	getAllSubscriptions()
+	getAllAggregates()
+	getAllDeviceSubscriptions()
+}
+func getAllSubscriptions() {
+	subscriptionsCache = make(map[string]evmodel.Subscription)
+	systemToSubscriptionsMap = make(map[string]map[string]bool)
+	aggregateIdToSubscriptionsMap = make(map[string]map[string]bool)
+	collectionToSubscriptionsMap = make(map[string]map[string]bool)
+	emptyOriginResourceToSubscriptionsMap = make(map[string]map[string]bool)
+	subscriptions, err := evmodel.GetAllEvtSubscriptions()
+	if err != nil {
+		l.Log.Error("Error while reading all subscription data ", err)
+		return
+	}
+	for _, subscription := range subscriptions {
+		var sub evmodel.SubscriptionResource
+		err = json.Unmarshal([]byte(subscription), &sub)
+		if err != nil {
+			continue
+		}
+		loadSubscriptionCacheData(sub)
+	}
+
+}
+
+func getAllDeviceSubscriptions() {
+	eventSourceToManagerIDMap = make(map[string]string)
+	t := time.Now()
+	defer l.Log.Debug("Time take to read complete aggregateToHost ", time.Since(t))
+	deviceSubscriptionList, err := evmodel.GetAllDeviceSubscriptions()
+	if err != nil {
+		l.Log.Error("Error while reading all aggregate data ", err)
+		return
+	}
+	for _, device := range deviceSubscriptionList {
+		devSub := strings.Split(device, "||")
+		if strings.Contains(devSub[0], "Collection") {
+			continue
+		} else {
+			updateCatchDeviceSubscriptionData(devSub[0], evmodel.GetSliceFromString(devSub[2]))
+		}
+	}
+}
+func updateCatchDeviceSubscriptionData(key string, originResources []string) {
+	systemId := originResources[0][strings.LastIndexByte(originResources[0], '/')+1:]
+	eventSourceToManagerIDMap[key] = systemId
+}
+
+func loadSubscriptionCacheData(sub evmodel.SubscriptionResource) {
+	if len(sub.EventDestination.OriginResources) == 0 && sub.SubscriptionID != "0" {
```

Comment:
Don't need to fill all of these separately. Just set the two attributes in a if/else block .

Reply:
done